Task1.SparseNumericMatrix/Entities/SparseMatrix.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Collections;
+using System.Text;
+
+namespace Task1.SparseNumericMatrix.Entities
+{
+    internal class SparseMatrix : IEnumerable<int> 
+    {
+        private struct Key
+        {
+            public int RowIndex { get; set; }
+            public int ColumnIndex { get; set; }
+            public Key(int rowIndex, int columnIndex)
+            {
+                RowIndex = rowIndex;
+                ColumnIndex = columnIndex;
+            }
+        }
+        public int RowCount { get; private set; }
+        public int ColumnCount { get; private set; }
+        private int _size => RowCount * ColumnCount;
+        private Dictionary<Key, int> _cells = new Dictionary<Key, int>();
+        public int this[int i, int j]
+        {
+            get
+            {
+                var key = new Key(i, j);
+                int value;
+                return _cells.TryGetValue(key, out value) ? value : default;
+            }
+            set
+            {
+                var key = new Key(i, j);
```

Comment:
Consider sparse matrix sized 1mln x 1mln with single non-null element at [0,0] index. 
It takes 4tb space and keeps single number. 

Do not keep zeroes

Reply:
Something simple as this works:
 set
            {
                var key = new Key(i, j);
                if (value != 0)
                {
                    _cells[key] = value;
                }
            }